pkg/activator/dedupe.go

```diff
@@ -13,30 +13,25 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
+
 package activator
 
 import (
 	"fmt"
 	"sync"
 )
 
-var shuttingDownError = activationResult{
-	endpoint: Endpoint{},
-	status:   Status(500),
-	err:      fmt.Errorf("activator shutting down"),
-}
-
-type activationResult struct {
-	endpoint Endpoint
-	status   Status
-	err      error
+var shuttingDownError = ActivationResult{
+	Endpoint: Endpoint{},
+	Status:   Status(500),
```

Comment:
If these are for HTTP can we use the enums from HTTP instead of numbers? e.g. `http.StatusInternalServerError`

Reply:
Done. I removed the special type and changed Status to be int and changed all places to use proper HTTP status codes.